File: ruby/lib/task_master/output_handlers/list_report/terminal_handler.rb
module TaskMaster
  module OutputHandlers
    module ListReport
      class TerminalHandler
        def generate_output(data)
          AsciiTable.new(headers, DataProcessor.new(data).process).draw
        end

        private

        def headers
          ["Task ID", "Task Name", "Assignee", "Due Date", "Status"]
        end

        class DataProcessor
          def initialize(data)
            @data = data
          end

          def process
            @data.map do |d|
              [d.id.to_s, d.name, d.assignee, d.due_on.to_s, d.status]
            end
          end
        end
      end
    end
  end
end

File: javascript/lib/outputHandlers/listReport/TerminalHandler.js
const path = require('path');
const AsciiTable = require(path.join(__dirname, '../../AsciiTable'));

class TerminalHandler {
	constructor() {
		this.AsciiTable = AsciiTable
		this.DataProcessor = class {
			constructor(data) {
				this.data = data
			}

			process() {
				return this.data.map(task => {
					return [
						String(task.id),
						task.name,
						task.assignee,
						this.#formatDateString(task.dueOn),
						task.status
					]
				})
			}

			#formatDateString(date) {
				return String(date.getFullYear() + '-' + (date.getMonth() + 1) + '-' + String(date.getDate()).padStart(2, '0'));
			}
		}
	}

	generateOutput(data) {
		return new this.AsciiTable(this.#headers(), new this.DataProcessor(data).process()).draw()
	}

	#headers() {
		return ["Task ID", "Task Name", "Assignee", "Due Date", "Status"]
	}
}

module.exports = TerminalHandler;

